scripts/install_third_party_libs.py

```diff
@@ -210,6 +219,74 @@ def ensure_system_python_libraries_are_installed(package, version):
     install_backend_python_libs.pip_install_to_system(package, version)
 
 
+def rewrite_android_proto_files():
+    """Edit all android proto files."""
+    # Since there is no simple configuration for imports when using protobuf to
+    # generate Python files we need to manually fix the imports.
+    # See: https://github.com/protocolbuffers/protobuf/issues/1491
+    protobuf_dir = (
+        pathlib.Path(
+            os.path.join(
+                common.THIRD_PARTY_DIR,
+                'oppia-proto-api-8f3cde883c31785438e80656a5b6bb26bd01b6a1'))
+            .glob('**/*.proto'))
+    for p in protobuf_dir:
+        if p.suffix == '.proto':
+            # Remove package statement.
+            # Example: 'package org.oppia.proto.v1.api'.
+            common.inplace_replace_file(
+                p.absolute(),
+                r'^package ([^\s]+)',
+                r'')
+            # Remove option statement.
+            # Example: 'option java_package = "org.oppia.proto.v1.api";'.
+            common.inplace_replace_file(
+                p.absolute(),
+                r'^option java_.+',
+                r'')
+            # Update import statement.
+            # Example: 'import "org/oppia/proto/v1/api/state.proto";' to
+            # 'import "state.proto";'.
+            common.inplace_replace_file(
+                p.absolute(),
+                r'^import (?!\"google)([^\s]+)[\\/]([^\s]+)',
+                r'import "\2')
+            # Remove all subpackage directories.
+            # Example: 'org.oppia.'.
+            common.inplace_replace_file(
+                p.absolute(),
+                r'org.oppia.*\.',
+                r'')
+
+
+def move_all_proto_files_to_third_party():
+    """Move all proto files from subdirectories
+    to the third_party folder.
+    """
+    oppia_proto_api_path = (
+        os.path.join(
+            common.THIRD_PARTY_DIR,
+            'oppia-proto-api-8f3cde883c31785438e80656a5b6bb26bd01b6a1'))
+    protobuf_dir = (
+        pathlib.Path(
+            oppia_proto_api_path).glob('**/*.proto'))
+
+    for p in protobuf_dir:
+        if p.suffix == '.proto':
+            shutil.move(
+                str(p.absolute()),
+                str(os.path.join(
+                    oppia_proto_api_path, os.path.basename(p.absolute()))))
+
+    # If there is any subfolder in the root folder from the
+    # oppia_proto_api repository in the third_party directory,
+    # then delete it. We are keeping all the proto files under
```

Comment:
Why is deleting the correct thing to do? I understand why we want to keep all the proto files in the root folder but doesn't that mean we should be throwing an exception if the files are not in the expected place, rather than deleting and (silently) continuing?

Reply:
Above this comment we had moved all the proto files, to the parent directory, keeping the empty folder doesn't make any sense to me. Also, it does confuse if someone in the future sees an empty directory.

Reply:
I didn't realize that the folder was empty. Why can't we delete it explicitly (by name)? The existing comment doesn't imply/explain that the directory is empty and doesn't make allowance for what happens if there are nonempty directories remaining at this point (e.g. suppose the code above changes in the future).

Reply:
I found another way of doing this. 
So, now I am not deleting this directory. The function `move_all_proto_files_to_third_party`, is all about moving all the files with the `.proto` extension to the root of the parent. 

Why I am not deleting the directory?
1. Firstly, we had already explained why we need to move the proto files to the root.
2. Secondly, If in the future someone checks this function, they will not get an idea from where these files are moved. So, keeping the directory intact will help them understand two things: A) from where we had moved these files, B) what exactly the proto issue because of which we had to move all the files to proto.